extensions-contrib/dropwizard-emitter/src/main/java/org/apache/druid/emitter/dropwizard/DropwizardEmitter.java

```diff
@@ -0,0 +1,168 @@
+    this.gagues = Collections.synchronizedMap(new GaugesCache<>(config.getMaxGaugeCount()));
+  }
+
+
+  @Override
+  public void start()
+  {
+    final boolean alreadyStarted = started.getAndSet(true);
+    if (!alreadyStarted) {
+      for (DropwizardReporter reporter : reporters) {
+        reporter.start(metricsRegistry);
+      }
+    }
+  }
+
+  @Override
+  public void emit(Event event)
+  {
+    synchronized (started) {
+      if (!started.get()) {
+        throw new RejectedExecutionException("Dropwizard emitter Service not started.");
+      }
+    }
+    if (event instanceof ServiceMetricEvent) {
+      ServiceMetricEvent metricEvent = (ServiceMetricEvent) event;
+      String host = metricEvent.getHost();
+      String service = metricEvent.getService();
+      String metric = metricEvent.getMetric();
+      Map<String, Object> userDims = metricEvent.getUserDims();
+      Number value = metricEvent.getValue();
+      ImmutableList.Builder<String> nameBuilder = new ImmutableList.Builder<>();
+      LinkedHashMap<String, String> dims = new LinkedHashMap<>();
+      final DropwizardMetricSpec metricSpec = converter.addFilteredUserDims(service, metric, userDims, dims);
+
+      if (metricSpec != null) {
+        if (config.getPrefix() != null) {
+          nameBuilder.add(config.getPrefix());
+        }
+        nameBuilder.add("metric=" + metric);
+        nameBuilder.add("service=" + service);
+        if (config.getIncludeHost()) {
+          nameBuilder.add("hostname=" + host);
+        }
+        dims.forEach((key, value1) -> nameBuilder.add(key + "=" + value1));
+
+        String fullName = StringUtils.replaceChar(Joiner.on(",").join(nameBuilder.build()), '/', ".");
+        updateMetric(fullName, value, metricSpec);
+      } else {
+        log.debug("Service=[%s], Metric=[%s] has no mapping", service, metric);
+      }
+    } else if (event instanceof AlertEvent) {
+      for (Emitter emitter : alertEmitters) {
+        emitter.emit(event);
+      }
+    } else {
+      throw new ISE("unknown event type [%s]", event.getClass());
+    }
+  }
+
+  private void updateMetric(String name, Number value, DropwizardMetricSpec metricSpec)
+  {
+    switch (metricSpec.getType()) {
+      case meter:
+        metricsRegistry.meter(name).mark(value.longValue());
+        break;
+      case timer:
+        metricsRegistry.timer(name)
+                       .update(value.longValue(), metricSpec.getTimeUnit());
+        break;
+      case counter:
+        metricsRegistry.counter(name).inc(value.longValue());
+        break;
+      case histogram:
+        metricsRegistry.histogram(name).update(value.longValue());
+        break;
+      case gauge:
+        Number prevVal = gagues.put(name, value);
+        // It is the first time we get this metric, register it with metricsRegistry
+        if (prevVal == null) {
+          metricsRegistry.register(name, (Gauge<Number>) () -> gagues.get(name));
+        }
+    }
```

Comment:
Please add `default` here to handle other cases. Probably log about the unknown type?

Reply:
added default, threw an exception instead.